torchaudio/functional/functional.py

```diff
@@ -1352,15 +1354,20 @@ def _get_sinc_resample_kernel(
     # they will have a lot of almost zero values to the left or to the right...
     # There is probably a way to evaluate those filters more efficiently, but this is kept for
     # future work.
-    idx = torch.arange(-width, width + orig_freq)
+    idx = torch.arange(-width, width + orig_freq, dtype=torch.float64)
 
     for i in range(new_freq):
         t = (-i / new_freq + idx / orig_freq) * base_freq
         t = t.clamp_(-lowpass_filter_width, lowpass_filter_width)
-        t *= math.pi
-        # we do not use torch.hann_window here as we need to evaluate the window
+
+        # we do not use built in torch windows here as we need to evaluate the window
         # at specific positions, not over a regular grid.
-        window = torch.cos(t / lowpass_filter_width / 2)**2
+        if resampling_method == "sinc_interpolation":
```

Comment:
Here is an idea. Since `"sinc_interpolation"` is not semantically correct, we deprecate it and add `"hann_window"` and change the default value to `"hann_window"`. (or was the default `square window`? I forgot)

Since there was no other choice before, we do not expect that a user would actually be passing `"sinc_interpolation"`, but if so we issue a warning to change it to `"hann_window"`, and in the future release, we can remove `"sinc_interpolation"` value. That way, the choices in the value of `resampling_method` makes more sense.

Reply:
After the offline conversation. I think we can leave this as-is. Sinc interpolation is used when apply kaiser window as-well, but as a function option, we can interpret it as "just sinc interpolation".